### PR TITLE
the queen can now prevent a xenomorph from communicating on the hivemind

### PR DESCRIPTION
**Pull request by harryob**
the queen's will will be expressed throughout the hivemind

this only persists until death so if that's a problem we can always move that around

:cl:
add: the queen can now mute insubordinate xenomorphs
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm [hunk 1]
```diff
@@ -469,6 +469,42 @@
...
+			break
+
+	var/reason = tgui_input_text(queen_user, "Why is [target_xeno] being excommunicated?")
+
+	if(!reason)
+		return
+
+	var/datum/weakref/target_ref = WEAKREF(target_xeno)
+
+	if(target_ref in queen_user.hive.muted_xenos)
```
Comment (TheGamerdk) [inline code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm:496]: This should probably be above the reason input

Comment (harryob) [reply code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm:496]: yep!

Comment (harryob) [reply code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm:491]: thanks btw, completely missed that somehow

File: code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm [hunk 2]
```diff
@@ -469,6 +469,42 @@
...
+		return
+
+	var/choice = tgui_input_list(queen_user, "Pick a Xenomorph:", "Toggle the Hivemind Connection of a Xenomorph", queen_user.hive.totalXenos)
+
+	if(!choice)
+		return
+
+	var/mob/living/carbon/xenomorph/target_xeno
+
+	for(var/mob/living/carbon/xenomorph/xeno in queen_user.hive.totalXenos)
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm:484]:
```suggestion
	for(var/mob/living/carbon/xenomorph/xeno as anything in queen_user.hive.totalXenos)
```

File: code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm [hunk 3]
```diff
@@ -469,6 +469,42 @@
...
+		if(html_encode(xeno.name) == html_encode(choice))
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm:485]: can two xenoes be named the same thing?

File: code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm [hunk 4]
```diff
@@ -469,6 +469,42 @@
...
+		return
+
+	var/reason = tgui_input_text(queen_user, "Why is [target_xeno] being excommunicated?")
+
+	if(!reason)
+		return
+
+	xeno_announcement("By [queen_user]'s will, [target_xeno] has been forbidden to communicate on the Hivemind!\n\nReason: [reason]", queen_user.hivenumber, title = SPAN_ANNOUNCEMENT_HEADER_BLUE("Excommunication"))
+	xeno_maptext("[target_xeno] is no longer able to communicate on the Hivemind.", "Excommunication", queen_user.hivenumber)
+	message_admins("[key_name_admin(queen_user)] has banished [key_name_admin(target_xeno)]. Reason: [reason]")
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm:504]: perhaps a better word than "banished" for hivemind muting

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm (hunk 3 above)
Comment (harryob) [reply code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm:485]: nope

**Review by Zonespace27 — APPROVED** [review]
Code good

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Spoke about changes on discord.

**Comment by Ultimate-Fluff** [comment]
> 

Yes, it could easily be abused, which is why we need the xeno to die first.

**Comment by morrowwolf** [comment]
Yes, unbanishing is being removed.

File: code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm [hunk 5]
```diff
@@ -469,6 +469,42 @@
...
+		if(html_encode(xeno.name) == html_encode(choice))
+			target_xeno = xeno
+			break
+
+	var/datum/weakref/target_ref = WEAKREF(target_xeno)
+
+	if(target_ref in queen_user.hive.muted_xenos)
+		queen_user.hive.muted_xenos -= target_ref
+		to_chat(queen_user, SPAN_XENONOTICE("[target_xeno] has been allowed to rejoin the Hivemind."))
+		to_chat(target_xeno, SPAN_XENOBOLDNOTICE("You are now able to communicate on the Hivemind."))
```
Comment (realforest2001) [inline code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm:494]: May be worth logging the unmute too

**Pull request closed without merging**